src/components/AllProducts.jsx

```diff
@@ -4,17 +4,22 @@ import { Link } from "react-router-dom";
 
 import { getAllProducts, formatCurrency } from "../api";
 
-export const AllProducts = () => {
+export const AllProducts = ({searchFilter=null}) => {
   const [productList, setProductList] = useState([]);
 
   const fetchData = async () => {
     const fetchedProducts = await getAllProducts();
-    setProductList(fetchedProducts);
+    setProductList(fetchedProducts.filter(product =>{
```

Comment:
How about we check for searchFilter first, before the .filter() method so we can bypass that if not required?